Cargo.toml

```diff
@@ -11,3 +11,13 @@ authors = [
 license = "ISC"
 repository = "https://github.com/rust-scraper/ego-tree"
 readme = "README.md"
+
+[features]
+serde = ["dep:serde"]
+
+[dependencies]
+serde = { version = "1.0.209", optional = true }
+
+[dev-dependencies]
+serde = "1.0.209"
+serde_json = "1.0.127"
```

Comment:
[`serde_test`](https://crates.io/crates/serde_test) might be preferable to using a specific format like JSON.

Reply:
Agreed, `serde_test` is the way to go here